src/prettytable/prettytable.py

```diff
@@ -74,6 +76,61 @@ def _get_size(text):
     return width, height
 
 
+##############################
+# ALIGNMENT VALIDATORS       #
+##############################
+
+
+def validate_align(val):
+    try:
+        assert val in ["l", "c", "r"]
+    except AssertionError:
+        raise ValueError(f"Alignment {val} is invalid, use l, c or r")
+
+
+def validate_valign(val):
+    try:
+        assert val in ["t", "m", "b", None]
+    except AssertionError:
+        raise ValueError(f"Alignment {val} is invalid, use t, m, b or None")
+
+
+def validate_header_align(val):
+    try:
+        assert val in ["l", "c", "r", None]
+    except AssertionError:
+        raise ValueError(f"Alignment {val} is invalid, use l, c, r or None")
+
+
+# A dict extension to manage field alignment. Used to:
+# Have global alingment sepreate from field alignments.
+# Validate values when setting for individual fields
+class Alignment(dict):
+    def __init__(self, align_type=ALIGN_TYPE_DEFAULT):
+        super().__init__()
+
+        self.align_type = align_type
+        if align_type == ALIGN_TYPE_VERTICAL:
+            self.alignment = "t"
+        elif align_type == ALIGN_TYPE_HEADER:
+            self.alignment = None
+        else:
+            self.alignment = "c"
+
+    def __setitem__(self, field_name, val):
+        if self.align_type == ALIGN_TYPE_VERTICAL:
+            validate_valign(val)
+        elif self.align_type == ALIGN_TYPE_HEADER:
+            validate_header_align(val)
+        else:
+            validate_align(val)
+        dict.__setitem__(self, field_name, val)
+
+    def __repr__(self):
+        dictrepr = dict.__repr__(self)
+        return dictrepr
```

Comment:
```suggestion
        return dict.__repr__(self)
```

Please also add a test case for this method.